### PR TITLE
Clarify span documentation regarding metadata

### LINKED ISSUE #77: span/3 StartMetadata and StopMetadata semantics
I was working on adding some [new telemetry events](https://github.com/commanded/commanded/pull/438) to Commanded to be used in the [bridge library](https://github.com/SimpleBet/opentelemetry_commanded) I'm developing for Otel. A point of confusion came for me when I needed to pass the metadata to `:telemetry.span/3`. In the PR, you'll see that I passed data metadata to the `StartMetadata`, but then passed an empty map for the `StopMetadata`.

After ruminating on this with @bryannaegele, it became obvious to me that I should be passing the same value to both start and stop. However, the documentation did not point me in this direction, and the API was surprising initially. I would have expected the `StopMetadata` to be merged into the `StartMetadata`. 

 I'm opening this issue to start a discussion on whether this needs a simple documentation update, or if the API needs to be reconsidered slightly.

### PR DESCRIPTION
**Pull request by davydog187**
Closes #77

### CODE REVIEW

**Review by bryannaegele — COMMENTED** [review]
We should include guidance that start metadata should be the basis of the stop metadata, i.e. it should be additive. Failure to do so removes valuable information to telemetry handlers which are only concerned about the stop event, e.g. metrics handlers.

**Comment by davydog187** [comment]
@bryannaegele added some more language

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: src/telemetry.erl [hunk 1]
```diff
@@ -155,14 +155,19 @@ execute(EventName, Measurements, Metadata) when is_map(Measurements) and is_map(
...
+%% used for metrics, can rely entirely on the `stop` event.
 %%
 %% A default span context is added to event metadata under the `telemetry_span_context` key if none is provided by
 %% the user in the `StartMetadata`. This context is useful for tracing libraries to identify unique
 %% executions of span events within a process to match start, stop, and exception events. Users
 %% should ensure this value is unique within the context of a process at a minimum if overriding this key and
 %% that the same value is provided to both `StartMetadata` and `StopMetadata`.
 %%
-%% For `telemetry' events denoting the <strong>start</strong> of a larger event, the following data is provided:
+%% For `telemetry` events denoting the <strong>start</strong> of a larger event, the following data is provided:
```
Comment (tsloughter) [inline src/telemetry.erl:170]: This should remain as `telemetry'.

Comment (davydog187) [reply src/telemetry.erl:170]: Is there some weird quirk of edoc that I'm missing?

Comment (tsloughter) [reply src/telemetry.erl:170]: That is just the edoc syntax for quoting.

I only now realize this PR has been merged, I wondered why I couldn't make my comment an actual suggested change :(

**Pull request merged**